### PR TITLE
feat: add issue auto labelling

### PR DESCRIPTION
**Pull request by caleb-at-pieces**
This adds an auto labelling system to our github issues

`.github/auto-labeling.yml` now run's 3 jobs to attach labels to issues. 

~~1. uses the community auto labeler action in order to add a 'bug' tag to issues that are created with the bug issue template. In order to extend this functionality as we add more templates, simply modify the `labeler.yml` file according to the same syntax it is currently in.~~ - this has been removed per our [earlier discussion](https://chat.google.com/room/AAAAjRPqM4g/EQ2mh4oE93w/EQ2mh4oE93w?cls=10)
2. will automatically attach a 'needs triage' label to all new issues
3. will attach a label according to the software the user chose when they filled out the issue template. this specific action heavily relies on each issue being the exact same format, so further issue templates should also have the same 'Software' dropdown field as the first thing within the issue. This workflow also uses npm and javascript, so I did need to create an NPM package within this repo. I think we could potentially move it somewhere else instead of being at the very top level of the repo, but we can discuss this later.

### CODE REVIEW

**Review by rosie-at-pieces — APPROVED** [review]

File: .github/workflows/auto-labelling.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Add Labels to new issues
+
+on:
+  issues:
+    types: [opened]
+
+jobs:
+  # label_issue_with_type:
```
Comment (tsavo-at-pieces) [inline .github/workflows/auto-labelling.yml:8]: Delete Dead Code and Add Comment With Link to https://github.com/actions/labeler

File: .github/workflows/auto-labelling.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            GH_REPO: ${{ github.repository }}
+            NUMBER: ${{ github.event.issue.number }}
+            LABELS: Needs triage
+  label_with_software:
+      runs-on: ubuntu-latest
+      steps:
+        - name: Checkout code
+          uses: actions/checkout@v2
+  
+        - name: Run labeling script
```
Comment (tsavo-at-pieces) [inline .github/workflows/auto-labelling.yml:33]: Line 33 - LGTM, Love the script

File: scripts/add_labels.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  const issue = await octokit.issues.get({
+    owner: 'pieces-app',
+    repo: 'support',
+    issue_number: issueNumber,
+  });
+
+  const issueBodyLines = issue.data.body.split('\n');
+
+  let software;
+  if (issueBodyLines[0] === '### Software') software = issueBodyLines[2];
```
Comment (tsavo-at-pieces) [inline scripts/add_labels.js:16]: potentially use includes and .at here

Comment (tsavo-at-pieces) [reply scripts/add_labels.js:16]: Not needed but a bit more graceful

**Review by tsavo-at-pieces — APPROVED** [review]
LGTM

**Pull request merged**